lib/Twig/Environment.php

```diff
@@ -297,20 +297,20 @@ public function getCacheFilename($name)
      *  * The currently enabled extensions;
      *  * Whether the Twig C extension is available or not.
      *
-     * @param string $name  The name for which to calculate the template class name
-     * @param int    $index The index if it is an embedded template
+     * @param string   $name  The name for which to calculate the template class name
+     * @param int|null $index The index if it is an embedded template
      *
      * @return string The template class name
      */
     public function getTemplateClass($name, $index = null)
     {
-        $key = $this->getLoader()->getCacheKey($name).'__'.implode('__', array_keys($this->extensions));
-
-        if (function_exists('twig_template_get_attributes')) {
-            $key .= '__cext';
```

Comment:
adding it optionally can create collision, e.g. an added twig extension whose name is "cext" would create the same key.